### PR TITLE
added custom LogicException to Connections commit method if nothing pers...

### LINKED ISSUE #356: Bad request exception on commit before persisting anything
If we try to commit before persisting anything (as we have situation when we do not always know that we persisted anything), we get bad request exception:
 ```
ElasticsearchParseException[Failed to derive xcontent from org.elasticsearch.common.bytes.BytesArray@1]
```

Error occurs when the following line returns empty result and we try to call bulk query with empty body:
https://github.com/ongr-io/ElasticsearchBundle/blob/9b18076e1af8f47fa61cc5e659effc6b3630ced4/Client/Connection.php#L156

Consider adding result check before calling bulk query or maybe throw custom exception?

### PR DESCRIPTION
**Pull request by mansimas**
...isted

Closes #356

### CODE REVIEW

File: Exceptions/EmptyBulkException.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\Exceptions;
```
Comment (saimaz) [inline Exceptions/EmptyBulkException.php]: Right name space would be `namespace ONGR\ElasticsearchBundle\Exception`

File: Client/Connection.php [hunk 2]
```diff
@@ -150,13 +151,17 @@ public function setBulkParams(array $params)
 
     /**
      * Flushes the current query container to the index, used for bulk queries execution.
+     *
+     * @throws EmptyBulkException
      */
     public function commit()
     {
+        if (!$this->bulkQueries) {
+            throw new EmptyBulkException('Nothing persisted to commit.');
```
Comment (saimaz) [inline Client/Connection.php]: Right message would be: `Bulk container is empty. Add some queries before commit.`

File: Tests/Unit/Client/ConnectionTest.php [hunk 3]
```diff
@@ -156,10 +156,34 @@ public function testGetClient()
...
+        $connection = new Connection($client, ['index' => 'foo']);
         $connection->setBulkParams(['refresh' => 'true']);
+        $connection->bulk('create', 'type', []);
+        $connection->commit();
+    }
+
+    /**
+     * Tests if exception is thown when only commit is made.
+     *
+     * @expectedException \LogicException
```
Comment (saimaz) [inline Tests/Unit/Client/ConnectionTest.php]: Should expect `EmptyBulkException`

File: Tests/Unit/Client/ConnectionTest.php [hunk 4]
```diff
@@ -156,10 +156,34 @@ public function testGetClient()
...
 
-        $connection = new Connection($client, []);
+        $connection = new Connection($client, ['index' => 'foo']);
         $connection->setBulkParams(['refresh' => 'true']);
+        $connection->bulk('create', 'type', []);
+        $connection->commit();
+    }
+
+    /**
+     * Tests if exception is thown when only commit is made.
```
Comment (saimaz) [inline Tests/Unit/Client/ConnectionTest.php]: Check spelling.

File: Tests/Unit/Client/ConnectionTest.php [hunk 5]
```diff
@@ -156,10 +156,34 @@ public function testGetClient()
...
+                0 => [
+                    'create' => [
+                        '_index' => 'foo',
+                        '_type' => 'type',
+                    ],
+                ],
+                1 => [],
+            ],
+        ];
+        $client->expects($this->once())->method('bulk')->with($result);
```
Comment (saimaz) [inline Tests/Unit/Client/ConnectionTest.php]: This is nothing related to empty bulk exception functionality. Please do it as a separate PR. Remove these lines with rebase to keep history clean.

**Comment by saimaz** [comment]
@mansimas any idea when you have time to fix your PR?

**Comment by mansimas** [comment]
I can do it today.

File: Tests/Unit/Client/ConnectionTest.php (hunk 5 above)
Comment (mansimas) [inline Tests/Unit/Client/ConnectionTest.php]: I cant revert this, because after my changes it fails.
